backend/tilavarauspalvelu/api/graphql/types/reservation/serializers/approve_serializers.py

```diff
@@ -55,9 +60,54 @@ def validate(self, data: ReservationApproveData) -> ReservationApproveData:
         self.instance.validators.validate_reservation_state_allows_approving()
         data["state"] = ReservationStateChoice.CONFIRMED
         data["handled_at"] = local_datetime()
+
+        if data["price"] == 0:
+            return data
+
+        if settings.PAYMENT_ORDERS_FOR_HANDLED_RESERVATIONS_ENABLED:
+            return self.validate_payment_order(data)
+
+        return data
+
+    def validate_payment_order(self, data: ReservationApproveData) -> ReservationApproveData:
+        data["should_delete_previous_payment_order"] = False
+
+        # In case reservation is approved -> set back to handled -> approved, there might be a payment order already
+        if hasattr(self.instance, "payment_order"):
+            payment_order = self.instance.payment_order
+
+            paid_in_webshop = payment_order.status in OrderStatus.paid_in_webshop
+            data["should_delete_previous_payment_order"] = not paid_in_webshop
+
+            if paid_in_webshop and payment_order.price_total != data["price"]:
+                msg = "Reservation already has a paid payment order with a different price."
+                raise ValidationError(msg, code=error_codes.RESERVATION_PRICE_CANNOT_BE_CHANGED)
+
```

Comment:
Can return **if** `paid_in_webshop` and skip validating the rest?
1. Additional data for creation of new payment order will not be needed.
2. Payment product (and pricing) thus also not necessary.
3. Come to think of it, the old tax percentage value should in that case also be retained, since that's what the reservation was already paid with (and the approval will confirm that price as standing).